src/sentry/seer/similarity/types.py

```diff
@@ -99,7 +103,11 @@ def from_raw(cls, project_id: int, raw_similar_issue_data: Mapping[str, Any]) ->
 
         if not parent_grouphash:
             # TODO: Report back to seer that the hash has been deleted.
-            raise SimilarGroupNotFoundError("Similar group suggested by Seer does not exist")
+            raise SimilarHashNotFoundError("Similar hash suggested by Seer does not exist")
+
+        if not parent_grouphash.group_id:
```

Comment:
I have confirmed that we have ~114M group hashes without a group ID.

```
select
  count(*)
from
  getsentry.sentry_grouphash as gh
where
  gh.group_id is null;
```

I have checked that a system hash and an in-app hash both point to a group_id:
<img width="952" alt="image" src="https://github.com/user-attachments/assets/4c6fb5d4-ebf2-42ed-9b8b-088b83f7a5da">

We have an example case in the Sentry project:
<img width="902" alt="image" src="https://github.com/user-attachments/assets/0674cf75-bfb3-4af7-9054-1366bad2f11f">

Reply:
Wow - IIRC there are ~800 M grouphashes, so that's ~12% - much higher than I would have expected!

I think the only way they could get into Seer is through the disappearing-events bug I mentioned in the PR description, as backfill should never send them (given that they have no group and therefore no event to get a stacktrace from). That means that the `SimilarHashMissingGroupError` added here should always indicate a disappearing event. So... we'll see how often it comes up.